src/fu-util-common.c

```diff
@@ -1286,6 +1304,117 @@ fu_util_device_to_string (FwupdDevice *dev, guint idt)
 	return g_string_free (str, FALSE);
 }
 
+const gchar *
+fu_util_plugin_flag_to_string (FwupdPluginFlags plugin_flag)
+{
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_UNKNOWN)
+		return NULL;
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_CLEAR_UPDATABLE)
+		return NULL;
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_USER_WARNING)
+		return NULL;
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_NONE) {
+		/* TRANSLATORS: Plugin is active and in use */
+		return _("Enabled");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_DISABLED) {
+		/* TRANSLATORS: Plugin is inactive and not used */
+		return _("Disabled");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_NO_HARDWARE) {
+		/* TRANSLATORS: not required for this system */
+		return _("Required hardware was not found");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_LEGACY_BIOS) {
+		/* TRANSLATORS: system is not booted in UEFI mode */
+		return _("Firmware can not be updated in legacy BIOS mode");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_CAPSULES_UNSUPPORTED) {
+		/* TRANSLATORS: capsule updates are an optional BIOS feature */
+		return _("UEFI capsule updates not available or enabled");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_UNLOCK_REQUIRED) {
+		/* TRANSLATORS: user needs to run a command */
+		return _("Firmware updates disabled; run 'fwupdmgr unlock' to enable");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_EFIVAR_NOT_MOUNTED) {
+		/* TRANSLATORS: the user is using Gentoo/Arch and has screwed something up */
+		return _("Required efivarfs filesystem was not found");
+	}
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_ESP_NOT_FOUND) {
+		/* TRANSLATORS: partition refers to something on disk, again, hey Arch users */
+		return _("UEFI ESP partition not detected or configured");
+	}
+
```

Comment:
The same strings are listed in libfwupdplugin.  Why have them twice - it's meaning needing to keep up with it twice.

Reply:
The one in libfwupdplugin isn't translated, there's no msgctx. The `fu_plugin_build_device_update_error()` is only really designed for when we set the update_error on the `FuDevice` and I don't think the strings have to be, or should be, necessarily the same. E.g. the one in `fu_plugin_build_device_update_error` might say `Device not updatable as UEFI Capsules are not available` and the other might say `UEFI capsule updates not available or enabled`

Reply:
So, I actually made them have the right prefix, e.g. `Not updatable as UEFI capsule updates not enabled`